CMakeLists.txt

```diff
@@ -69,12 +70,31 @@ else()
     message(STATUS "Configuring Hyprland in Release with CMake")
 endif()
 
-include_directories(
-  .
-  "subprojects/wlroots/include/"
-  "subprojects/wlroots/build/include/"
-  "subprojects/udis86/"
-  "protocols/")
+set(WLROOTS_SOVERSION 12032)
+
+find_program(MESON_EXE NAMES meson)
+find_program(NINJA_EXE NAMES ninja)
+find_program(SED_EXE NAMES sed)
+ExternalProject_Add(wlroots_build
+    PREFIX wlroots
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/subprojects/wlroots
+    PATCH_COMMAND ${SED_EXE} -E -i -e "s/(soversion = 12)([^032]|$$)/soversion = ${WLROOTS_SOVERSION}/g" ${CMAKE_SOURCE_DIR}/subprojects/wlroots/meson.build
+    CONFIGURE_COMMAND ${MESON_EXE} setup --prefix ${CMAKE_CURRENT_BINARY_DIR}/wlroots . ${CMAKE_SOURCE_DIR}/subprojects/wlroots
+    BUILD_COMMAND ${NINJA_EXE} -C .
+    INSTALL_COMMAND ${NINJA_EXE} -C . install)
```

Comment:
we should not install wlroots. We only copy the `.so`.

Reply:
The install command is only used to copy wlroots (headers and lib) files to another location, it doesn't get installed in the system.

Reply:
uhh, `ninja -C build install` would install the headers to `/usr` which we don't want, no?

Reply:
I don't believe this is resolved

Reply:
The setup command sets the prefix to a location within the cmake build folder.

Reply:
right, my bad